picard/tagger.py

```diff
@@ -355,20 +356,44 @@ def add_files(self, filenames, target=None):
             if ignoreregex is not None and ignoreregex.search(filename):
                 log.info("File ignored (matching %s): %s" % (pattern, filename))
                 continue
-            if filename not in self.files:
+            if filename not in tmp_files:
                 file = open_file(filename)
                 if file:
-                    self.files[filename] = file
-                    new_files.append(file)
-        if new_files:
+                    tmp_files[filename] = file
+        if tmp_files and self.check_load(tmp_files):
+            new_files = []
+            for filename in sorted(tmp_files):
+                file = tmp_files[filename]
+                self.files[filename] = file
+                new_files.append(file)
             log.debug("Adding files %r", new_files)
-            new_files.sort(key=lambda x: x.filename)
             if target is None or target is self.unmatched_files:
                 self.unmatched_files.add_files(new_files)
                 target = None
             for file in new_files:
                 file.load(partial(self._file_loaded, target=target))
 
+    def check_load(self, new_files):
+        #Load large amounts of files
+        if len(new_files) > config.setting["file_import_threshold"]:
+            QMessageBox = QtGui.QMessageBox
+
+            msg = QMessageBox()
+            msg.setIcon(QMessageBox.Warning)
+            msg.setWindowModality(QtCore.Qt.WindowModal)
+            msg.setWindowTitle(_(u"Large File Load"))
+            msg.setText(_(u"Are you sure you want to load %s files? Picard may "
```

Comment:
`%s` should be replaced by `%d` here, value is an integer.
Also, while it is unlikely you have 0 or 1 file here, plural forms depend on languages, so you have to do something like https://github.com/metabrainz/picard/blob/1d1579130b02f9ecb496d7ccae925e77d4269ac5/picard/tagger.py#L398-L402

It will allow translators to provide proper translations.